### DISCUSSION TITLE
Feature idea: automatically play folders for unknown cards

### DISCUSSION THREAD

**Discussion by jeverling** [MiczFlor/RPi-Jukebox-RFID#1699] [Ideas]
Hi all, I wanted to allow others, potentially not very technical minded people, to add new music to a phoniebox.
With the webinterface it's not that hard to register new cards, but I think it would be nice to have a way where not even that is necessary.

What I did now, is to register 50 cards and create 50 folders named like the card IDs inside the audio folder. Now someone just has to pick one of the pre-registered cards, put music into the corresponding folder and it will work.

I think it would be really nice, to have an optional feature where the software checks if there is a folder on the filesystem inside the audio folder corresponding to an unknown ID, and plays it if that is the case. This way people could add new cards by simply creating a folder named like the card ID in the samba share, without having to understand the process of registering cards (either in the webinterface or manually).

**Comment by pabera** [comment]
I like this idea a lot, and it should be fairly simple to implement

**Comment by ghost** [comment]
Let's check how it can be integrated as part of the current changes to the player interface.

**Comment by MiczFlor** [comment]
Hi @jeverling 
in version 2.3 -- which is the current stable version -- this is exaclty how Phoniebox behaves. 

1. check if the card triggers a command, like "increase volume"
2. if not: check if a card with this ID is already registered
3. if not: create a shortcut automatically with a link to a folder by the same name as the card ID

See here:
https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/b3e91ae567cd3d089708f5b9d1a5f634bb41e8db/scripts/rfid_trigger_play.sh#L325

**Reply by jeverling** [reply]
I didn't know that, that's great to hear! :)
